database/src/user.rs

```diff
@@ -98,11 +134,52 @@ pub enum Role {
     User,
 }
 
+#[derive(Copy, Clone, Debug, Deserialize, Eq, PartialEq, Serialize, sqlx::Type)] // Does this need to be clone?
+#[sqlx(transparent)]
+pub struct UserID(pub(crate) i64);
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
+pub struct Roles(pub Vec<String>);
```

Comment:
Wonder if it's possible for this to be a enum. I haven't thought yet about how permissions will work in the future, so we can leave like this for now, but it's still something we should think about

Reply:
I had that thought, but it might be much more difficult to handle client side if it were a bitfield (On second thought - Does client-side even care about roles? If not, I guess it'd be pretty easy)

Reply:
It will in the future. Based on the roles we will disable certain UI elements.